enhancements/bootimages.md

```diff
@@ -0,0 +1,163 @@
+```
+
+would output data similar to the [aws rhcos.json data](https://github.com/openshift/installer/blob/2055609f95b19322ee6cfdd0bea73399297c4a3e/data/data/rhcos.json#L2) so that AWS UPI installations could use it programatically.
+
+Other output proposals
+
+- `metal`, `openstack`, `qemu`: What's available today for 4.3
+- `aws-snapshot` to handle [private AWS regions](https://github.com/openshift/os/blob/411d1f5943ea23f2de7e4f7a04ab0bb185fd9586/FAQ.md#q-how-do-i-get-rhcos-in-a-private-ec2-region).
+- `baremetal-iso` generates a "Live" ISO as is used in Fedora CoreOS and points people at how to do [Ignition injection](https://github.com/coreos/coreos-installer/blob/3652e6a767bad593b1b898f239e41bc11a83ab8f/src/iso.rs#L28)
+- `baremetal-pxe` outputs the split kernel/initramfs suitable for PXE
+
+### Non-Goals
+
+#### Replacing the default in-place update path
+
+In-place updates as [managed by the MCO](https://github.com/openshift/machine-config-operator/blob/master/docs/OSUpgrades.md) today works fairly seamlessly.  We can't require that everyone fully reprovision a machine in order to do in-place updates - that makes updates *much* more expensive.  It implies re-downloading all container images, etc.
+
+#### Exposing a general purpose "OS build" tool
+
+This proposal includes shipping a subset of https://github.com/coreos/coreos-assembler/ - but that's not the same as supporting custom "CoreOS-style" builds for non-OpenShift use cases.
+
+## Proposal phase 1: bootimage.json in release image
+
+First, the RHCOS build process is changed to inject the current coreos-assembler `meta.json` output for the build into `machine-os-content`.  This aims to move the "source of truth" for cluster bootimages into the release image.  Nothing would use this to start - however, as soon as we switch to a machine API provisioned control plane, having that process consume this data would be a natural next step.
+
+In fact, we could aim to switch to having workers use the `bootimage.json` from the release payload immediately after it lands.  A downside is this would open up potential for drift between the bootstrap+controlplane and workers.
+
+## Proposal phase 2: oc adm release generate-bootimage
+
+The implementation of this is basically shipping a subset of [coreos-assembler](https://github.com/coreos/coreos-assembler) as part of the OpenShift release payload, and teaching `oc` how to invoke `podman` to run it.
+
+The `generate-bootimage` implementation would download the `machine-os-bootimage-generator` container image along with the existing `machine-os-content` container image (OSTree repository), and effectively run the [create_disk.sh](https://github.com/coreos/coreos-assembler/blob/30fbac4e176c7936362efbd647c8199d927e593c/src/create_disk.sh) process or [buildextend-installer](https://github.com/coreos/coreos-assembler/blob/30fbac4e176c7936362efbd647c8199d927e593c/src/cmd-buildextend-installer) for live media, etc.
```

Comment:
Why do this at all?  If we're already pulling all the binaries to build the image, why not just publish an image that contains a ISO or whatever format?

Reply:
Good question.  I think the basic answer is that if we e.g. ship a container image wrapping each pre-built image, that's at least 700MB per image and there are like 7-8 or more images.  That's...a lot.  So things like `oc adm release mirror` would have to learn how to filter them.  It'd make the OpenShift release process to push them much slower.

It just seems more tenable to me to make this dynamic.

Reply:
>  So things like oc adm release mirror would have to learn how to filter them

That seems like a better approach.

Reply:
Can you elaborate a bit more on why?  Just in terms of avoiding needing a new nontrivial container image?  Needing to ship code that was formerly "behind the firewall" to run on premise?  Something else?

Reply:
On the bit level, we'll know the image they are booting is the one we built, we don't have to worry about a bug in the builder producing something incorrect in their environment.  It's also one less moving piece we need to worry about in-cluster.  My preference is to add nothing to the cluster that doesn't need to 100% need to be there.